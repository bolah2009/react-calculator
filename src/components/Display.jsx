import React from 'react';
import PropTypes from 'prop-types';
import styles from '../css/modules/Display.module.css';

const Display = ({ result }) => (
  <div className={`d-flex col jc-c ${styles.display}`}>{result}</div>
);

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
