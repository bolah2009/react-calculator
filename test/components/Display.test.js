import React from 'react';
import renderer from 'react-test-renderer';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Display from '../../src/components/Display';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Display component', () => {
  test('should render with default result value', () => {
    const display = renderer.create(<Display />);
    expect(display.toJSON()).toMatchSnapshot();
  });

  test('should render with an assigned result value', () => {
    const display = renderer.create(<Display result="20" />);
    expect(display.toJSON()).toMatchSnapshot();
  });
});

it('renders with or without result property', () => {
  act(() => {
    render(<Display />, container);
  });
  expect(container.textContent).toBe('0');

  act(() => {
    render(<Display result="500" />, container);
  });
  expect(container.textContent).toBe('500');
});
