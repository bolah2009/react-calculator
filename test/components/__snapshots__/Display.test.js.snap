// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component should render with an assigned result value 1`] = `
<div
  className="d-flex col jc-c display"
>
  <div
    className="currentButton"
  >
    
  </div>
  <div
    className="mainDisplay"
    id="display"
  >
    20
  </div>
</div>
`;

exports[`Display component should render with default result value 1`] = `
<div
  className="d-flex col jc-c display"
>
  <div
    className="currentButton"
  >
    
  </div>
  <div
    className="mainDisplay"
    id="display"
  >
    0
  </div>
</div>
`;
